src/HIE/Bios/Cradle.hs

```diff
@@ -345,6 +346,16 @@ cabalAction work_dir mc l _fp = do
 removeInteractive :: [String] -> [String]
 removeInteractive = filter (/= "--interactive")
 
+-- Strip out any ["+RTS", ..., "-RTS"] sequences in the command string list.
+removeRTS :: [String] -> [String]
+removeRTS  = snd . foldr processPath (False, [])
+  where processPath p (inRTS, paths) =
+          case (inRTS, p) of
+            (_, "+RTS") -> (False, paths)
+            (_, "-RTS") -> (True, paths)
+            (True, _)   -> (inRTS, paths)
+            (_, p')     -> (False, p' : paths)
+
```

Comment:
Sorry for a drive-by comment. Would code like this be simpler?

```haskell
removeRTS :: [String] -> [String]
removeRTS [] = []
removeRTS ("+RTS" : xs) =
  case dropWhile (/= "-RTS") xs of
    [] -> []
    (_ : ys) -> removeRTS ys
```

Reply:
If i'm understanding the proposal, I *think* your code will work with the addition of this case after your code.

```
removeRTS (y:ys) = y : removeRTS ys
```

I like the readability of what you wrote a lot and can change/test it again later. Thank you for having a look!

Reply:
Oops. Yeah, I forgot to add that case :)

Reply:
A new version is up.